### ISSUE TITLE
Empty output with pipeline.finders

### ISSUE THREAD

**Issue by addremove** [cyberdelia/django-pipeline#418]
The setting below generates empty JS/CSS files in `STATIC_ROOT`

    STATICFILES_FINDERS = ( 
        'pipeline.finders.FileSystemFinder',
        'pipeline.finders.AppDirectoriesFinder',
        'pipeline.finders.CachedFileFinder',
        'pipeline.finders.PipelineFinder',
    )

These work, but they also copy the source JS/CSS when I run `collectstatic`. 

    STATICFILES_FINDERS = ( 
        'django.contrib.staticfiles.finders.FileSystemFinder',
        'django.contrib.staticfiles.finders.AppDirectoriesFinder',
        'pipeline.finders.PipelineFinder',
    )

**Comment by akszydelko** [comment]
I had the same problem, but I found this thread after some debugging and I found reason for that behaviour.

pipeline's finders are doing a good job but only when `collectstatic` is used. During which all files are ignored (not copied to `STATIC_ROOT `) and only files from compression process are saved there (to `STATIC_ROOT `). By this it works perfect no crap in static root only results of compression process. But...

In debug mode files are not compressed, but should be copied as they are to static root (compilation of less/sass is made later on those copied files). But because of the ignored list from finder classes, static files are not copied to static root for further usage, so `FileSystemFinder` can not find them.

This is the course of problems:
https://github.com/cyberdelia/django-pipeline/blob/master/pipeline/finders.py#L69-L117 
https://github.com/cyberdelia/django-pipeline/blob/master/pipeline/glob.py#L29 

Because of the above this for is never entered, and paths variable is never filled. The path variable is essential for ruther procedures to render script tags.
https://github.com/cyberdelia/django-pipeline/blob/master/pipeline/packager.py#L26-L28

So basicly I see two options
1) Somehow use pipeline's finder only for `collectstatic` purpose.
2) Exclude files from pipeline configuration to be ignored (but this will also copy them while using `collectstatic` and they will be useless because on production compressed files are used).

Unfortunately I don't have time right now for further digging or creating fixing pull request ;/, but I think my observations will help.

I am also thinking about inappropriate setting configuration as @addremove suggest, because file finders are able to find those files in static dir inside my apps, so I am not sure if we should blame django for that.

**Comment by akszydelko** [comment]
Right in time after I wrote previous comment it hit me...
The workaround solution is very simple.

On production use this: 
```
STATICFILES_FINDERS = ( 
    'pipeline.finders.FileSystemFinder',
    'pipeline.finders.AppDirectoriesFinder',
    'pipeline.finders.CachedFileFinder',
    'pipeline.finders.PipelineFinder',
)
```

But for local development this:
```
STATICFILES_FINDERS = ( 
    'django.contrib.staticfiles.finders.FileSystemFinder',
    'django.contrib.staticfiles.finders.AppDirectoriesFinder',
    'pipeline.finders.PipelineFinder',
)
```

But of course it would be nice to have it defined only once.

**Comment by bitcity** [comment]
@cyberdelia  Any comment on why `pipeline.finders.FileSystemFinder` & `pipeline.finders.AppDirectoriesFinder` is generating empty files?

**Comment by bitcity** [comment]
Probably related https://github.com/cyberdelia/django-pipeline/issues/398

**Comment by bitcity** [comment]
Debugging further, I found `1.3.7` works with `STATICFILES_FINDERS` set to `pipeline.finders.*`. All the versions after that generate empty output for JS files.

**Comment by qcaron** [comment]
So I changed the finders to the foloowing in my **development environment**:
```python
STATICFILES_FINDERS = ( 
    'django.contrib.staticfiles.finders.FileSystemFinder',
    'django.contrib.staticfiles.finders.AppDirectoriesFinder',
    'pipeline.finders.PipelineFinder',
)
```

But because I manage front-end packages with Bower then I get a lot of directories and files in my static tree (for example, the `src` directory for Bootstrap is copied in the Django project STATIC_URL but not removed after compression tasks).

Please see https://github.com/cyberdelia/django-pipeline/issues/482 for more information about polluted STATIC_URL directory and slow page load.

If someone could answer to give advice or review the ticket that would be great. It seems there are only two of us having this problem or at least speaking about it.

**Comment by SverkerSbrg** [comment]
Has there been any updates on this issue. I'm running in to a similar problem arising when pipeline tries to compile (sass) files during development. Some digging reveals pipeline looking for the sass files in the static root instead of static dirs resulting in errors when sass does not find the file.

Am I doing anything wrong? Is there any suggested fixes? Or is it relevant to fork? In this case does any one have a good place to start at?

**Comment by samdieck** [comment]
I have the same problem.... pipeline.finders.* generates empty files while django.contrib.staticfiles.finders.* do generate files correctly but also copies every file in the static dir.

**Comment by SverkerSbrg** [comment]
@samdieck look att #499 it broke things for me

**Comment by hheimbuerger** [comment]
I have that problem, too. Trying to wrap my head around the issue and looking for options.

In the meantime, isn't the first bug here that "invalid" paths in `source_filenames` do not yield an error message? Is there any reason to stay silent about such a critical error?

Then at least noticing this problem and troubleshooting it would already get a lot easier.

**Comment by MrCsabaToth** [comment]
+1

**Comment by belugame** [comment]
+1

**Comment by mcanaves** [comment]
+1

**Comment by tehfink** [comment]
+1
